main.cpp

```diff
@@ -312,71 +237,14 @@ struct Numeric
     }
 };
 //----------------------------------------------
-template<>
-struct Numeric<double>
-{
-    using Type = double;
-    explicit Numeric(Type val) : value(std::make_unique<Type>(val)) {}
-
-    Numeric& operator+=( Type rhs )
-    {
-        *value += rhs; 
-        return *this;
-    }
-    Numeric& operator-=( Type rhs )
-    {
-        *value -= rhs; 
-        return *this;
-    }
-    Numeric& operator*=( Type rhs )
-    {
-        *value *= rhs; 
-        return *this;
-    }
-
-    Numeric& operator/=( Type rhs )
-    {
-        if(rhs == 0.0)
-        {
-            std::cout << "warning: floating point division by zero!" << std::endl; 
-        }
-
-        *value /= rhs;
-        return *this;
-    }
-
-    operator Type() const { return *value; }
-
-    Numeric& pow(const Type& t)
-    {
-        return powInternal(t);
-    }
-
-    template<typename T>
-    Numeric& apply(T freeFunc)
-    {
-        freeFunc(*value);
-        return *this;
-    }
-
-private:
-    std::unique_ptr<Type> value;
-    Numeric& powInternal(const Type& d)
-    {
-        *value = static_cast<Type>(std::pow( *value, d ));
-        return *this;
-    }
-};
 
 struct Point
 {
     template<typename T>
     Point(T m_x, T m_y) : x(static_cast<float>(m_x)), y(static_cast<float>(m_y)) {}
+    template<typename TRef>
+    Point(TRef& m_x, TRef& m_y) : Point(m_x, m_y) {}
```

Comment:
If you aren't modifying the object, pass by const-ref.
this second constructor is not needed, once you change the first to take const-ref arguments.
All that it currently does is create copies, because the first constructor takes the parameters by value instead of by const-ref
- [ ] revise